hs_rest_api/resources/file_metadata.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework import serializers
+
+from hs_core.models import ResourceFile, Coverage
+
+
+# TODO: Once we upgrade past Django Rest Framework 3.3, this won't be necessary
+class JSONSerializerField(serializers.Field):
+    """ Serializer for JSONField -- required to make field writable"""
+
+    def to_internal_value(self, data):
+        return data
+
+    def to_representation(self, value):
+        return value
+
+
+class FileMetaDataSerializer(serializers.Serializer):
+    title = serializers.CharField(required=False)
+    keywords = JSONSerializerField(required=False)
+    spatial_coverage = JSONSerializerField(required=False)
+    extra_metadata = JSONSerializerField(required=False)
+    temporal_coverage = JSONSerializerField(required=False)
+
+
+class FileMetaDataRetrieveUpdateDestroy(generics.RetrieveUpdateDestroyAPIView):
+    serializer_class = FileMetaDataSerializer
+    allowed_methods = ('GET', 'PUT')
+
+    def get(self, request, pk, file_id):
+        """ Get a resource file's metadata. """
+        resource_file = get_object_or_404(ResourceFile, id=file_id)
+
+        if resource_file.metadata is None or not resource_file.has_logical_file:
+            return Response({}, status=404)
+
+        title = resource_file.metadata.logical_file.dataset_name \
+            if resource_file.metadata.logical_file else ""
+        keywords = resource_file.metadata.keywords \
+            if resource_file.metadata else []
+        spatial_coverage = resource_file.metadata.spatial_coverage.value \
+            if resource_file.metadata.spatial_coverage else {}
+        extra_metadata = resource_file.metadata.extra_metadata \
+            if resource_file.metadata else []
+        spatial_coverage = resource_file.metadata.temporal_coverage.value if \
+            resource_file.metadata.temporal_coverage else {}
+
+        # TODO: How to leverage serializer for this?
+        return Response({
+            "title": title,
+            "keywords": keywords,
+            "spatial_coverage": spatial_coverage,
+            "extra_metadata": extra_metadata,
+            "temporal_coverage": spatial_coverage
+        })
+
+    def put(self, request, pk, file_id):
+        """ Update a resource file's metadata """
+        file_serializer = FileMetaDataSerializer(request.data)
+
+        title = file_serializer.data.pop("title", "")
+        resource_file = ResourceFile.objects.get(id=file_id)
+        resource_file.metadata.logical_file.dataset_name = title
+        resource_file.metadata.logical_file.save()
+
+        spatial_coverage = file_serializer.data.pop("spatial_coverage", None)
+        if spatial_coverage is not None:
+            if resource_file.metadata.spatial_coverage is not None:
+                Coverage.update(resource_file.metadata.spatial_coverage.id,
+                                _value=json.dumps(spatial_coverage))
+            # elif resource_file.metadata.spatial_coverage is None:
```

Comment:
@aphelionz See the AbstractFileMetaData class which have these CRUD functions:

https://github.com/hydroshare/hydroshare/blob/develop/hs_file_types/models/base.py#L332

https://github.com/hydroshare/hydroshare/blob/develop/hs_file_types/models/base.py#L347

https://github.com/hydroshare/hydroshare/blob/develop/hs_file_types/models/base.py#L347

so to create a coverage element of type spatial:
```resource_file.metadata.create_element('coverage', **kwargs)```
as part of kwargs you need to pass type='box' or type='point' and the corresponding location values
In case of temporal coverage: type='period'